store.go

```diff
@@ -240,3 +242,84 @@ func (s *SafeConfigStore) Size() int {
 	defer s.mu.Unlock()
 	return s.c.Size()
 }
+
+// LoggableConfigStore implements the ConfigStore interface and provides a store
+// a field for reference to a logging mechanism to log on reads/writes
+type LoggableConfigStore struct {
+	c   ConfigStore
+	log DefaultLogger
```

Comment:
I think this field should be a `Logging` type, right? so users can swap out the logging implementation?

Reply:
@moogar0880 they pass a logging interface which it set to the field of a `DefaultLogger`

Reply:
@lowellmower right, but if a caller was using a differen't logging implementation then wouldn't that mean that they don't want the `DefaultLogger` and instead want to use whatever they have that implements the `Logging` interface?

Reply:
DefaultLogger is merely the wrapper around the logging interface which provides some formatting, e.g. time stamp and log line suffix. This _could be_ just a logging interface if preferred, only have to move the log line formatting somewhere.

Reply:
Yeah I feel like it would be easier to do that in the logging interface. Means there's one less thing to wrap

Reply:
Ok, only issue I see with just having the interface and not a string which implements the interface being we have no place to put the default log line configuration, i.e. spacing, time stamp, and application name

Reply:
Moved the formatting method to a package level function. Let me know what you think.